msgraph/client.go

```diff
@@ -149,83 +150,77 @@ func (c Client) performRequest(req *http.Request, input HttpRequestInput) (*http
 		}
 	}
 
-	var attempts, backoff, multiplier int64
-	for attempts = 0; attempts < requestAttemptsForRateLimiting; attempts++ {
-		// sleep after the previous failed attempt
-		if attempts > 0 {
-			time.Sleep(time.Duration(backoff))
-		}
-
-		// default exponential backoff
-		multiplier++
-		backoff = int64(retryBackoffInitialDelay) * backoffPower(2, multiplier)
-		if cap := int64(retryBackoffDelayCap); backoff > cap {
-			backoff = cap
-		}
-
-		req.Body = io.NopCloser(bytes.NewBuffer(reqBody))
-		resp, err = c.httpClient.Do(req)
-		if err != nil {
-			return nil, status, nil, err
-		}
+	c.retryableClient.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
```

Comment:
Just a thought - is it possible / worthwhile to export this default retry function somehow? I'm just thinking, what if I want to pass in a custom HTTP client, but I still want to use the same retry behaviour? That way maybe I could do something along these lines:

```go
r := retryablehttp.NewClient()
r.HTTPClient = wrappedClient
r.CheckRetry = msgraph.DefaultCheckRetry

graphClient := msgraph.NewApplicationsClient(tenantID, r.StandardClient())
```

Reply:
That would be a nice feature, but the problem is that the default retry behavior sources a `ConsistencyFailureFunc` directly from the input struct for each request, so it can be determined whether the failure was due to replication delay within Azure AD. If you wanted to get the generic retry behavior (e.g. 429 backoff, 5xx retries) you can just use `retryablehttp` as it's providing that functionality here.